### ISSUE TITLE
conda build fails when multiple python/numpy versions are specified

### ISSUE THREAD

**Issue by bryanwweber** [conda/conda-build#1316]
Hi! I'm using conda-build 2.0.0 for my package, and I'm trying to build for multiple Python/Numpy versions in a single command. I didn't see anything in the CHANGELOG that this functionality was removed, but if it was, my mistake. The specific error messages are
```
conda build cantera --python=2.7 --python=3.4 --python=3.5 --numpy=1.11
Traceback (most recent call last):
  File "/home/bryan/software/anaconda3/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 234, in main
    execute(sys.argv[1:])
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 180, in execute
    config = Config(**args.__dict__)
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/config.py", line 36, in __init__
    self.set_keys(**kwargs)
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/config.py", line 58, in set_keys
    self.CONDA_PY = int(env('python', "%s%s" % (sys.version_info.major, sys.version_info.minor))
AttributeError: 'list' object has no attribute 'replace'
```
and for Numpy:
```
conda build cantera --python=3.5 --numpy=1.11 --numpy=1.10
Traceback (most recent call last):
  File "/home/bryan/software/anaconda3/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 234, in main
    execute(sys.argv[1:])
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/cli/main_build.py", line 195, in execute
    set_language_env_vars(args, parser, config=config, execute=execute)
  File "/home/bryan/software/anaconda3/lib/python3.5/site-packages/conda_build/render.py", line 43, in set_language_env_vars
    execute(args, parser, config)
TypeError: execute() takes 1 positional argument but 3 were given
```
I get the same error on Linux (Ubuntu) and OS X. My conda-info:
```
conda info
Current conda install:

             platform : linux-64
        conda version : 4.1.11
    conda-env version : 2.5.2
  conda-build version : 2.0.0
       python version : 3.5.2.final.0
     requests version : 2.11.1
     root environment : /home/bryan/software/anaconda3  (writable)
  default environment : /home/bryan/software/anaconda3
     envs directories : /home/bryan/software/anaconda3/envs
        package cache : /home/bryan/software/anaconda3/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/conda-forge/linux-64/
                        https://conda.anaconda.org/t/<TOKEN>/conda-forge/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/bryanwweber/channel/devel/linux-64/
                        https://conda.anaconda.org/t/<TOKEN>/bryanwweber/channel/devel/noarch/
                        https://conda.anaconda.org/t/<TOKEN>/bryanwweber/linux-64/
                        https://conda.anaconda.org/t/<TOKEN>/bryanwweber/noarch/
          config file : /home/bryan/.condarc
         offline mode : False
    is foreign system : False
```

The recipe is located here: https://github.com/Cantera/conda-recipes although I'm having the same issue with other recipes as well. Any help is appreciated. Thank you!

**Comment by msarahan** [comment]
Yeah, sorry I broke that.  I forgot to add a note to the release notes.  What I recommend instead is conda-build-all: https://github.com/SciTools/conda-build-all

Long-term, conda-build will support this in a much more flexible way ( #1142 ), but until then, I'm sorry, this will be broken for a while.

**Comment by bryanwweber** [comment]
@msarahan OK thank you for the update! I actually have another question that I think is related to #1142. In the recipe I referenced above, we use SCons to build the C extension to the Python module, and so we set the package version in the SConstruct file so that it is the same in all the interfaces we have. Thus, the version isn't written to the setup.py file until the build is finished, so we can't use `load_setup_py_data` in the jinja template because there is no version in setup.py at render time. What would you suggest we do to set the version in a single place? Should we just deal with setting the version in SConstruct and meta.yaml separately (and risk mis-matching versions) until the changes in #1142 come through? 

One way that we came up with is to patch SConstruct so that it accepts the version from an external configuration file and write the value of `PKG_VERSION` to that file. However, the patch is really fragile so this seems suboptimal. Is there a better way we can use right now?

**Comment by msarahan** [comment]
I haven't used SCons much, but I'm in favor of adding support to conda-build for anything you need.  It would probably be something very much like ```load_setup_py_data``` but would parse your scons file where the version lives.

Essentially, if your version is somewhere in your source code, be it as a git tag, a setup.py version, or whatever, then there should be a way to get it pre-build, and I'll work to support that.

**Comment by bryanwweber** [comment]
OK thank you for the support! I'll talk with @speth offline and see if we can come up with a useful way to extract the information from the file.

**Comment by bryanwweber** [comment]
Closed by #1443
